pipeline/src/node_properties/transform_properties.rs

```diff
@@ -0,0 +1,114 @@
+use serde::{Deserialize, Serialize};
+use std::str::FromStr;
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub enum FlipDirection {
+    Horizontal,
+    Vertical,
+}
+
+#[derive(Default, Debug, Clone, PartialEq)]
+pub struct Crop {
+    /// Number of pixels to crop from the top.
+    pub top: usize,
+
+    /// Number of pixels to crop from the bottom.
+    pub bottom: usize,
+
+    /// Number of pixels to crop from the left.
+    pub left: usize,
+
+    /// Number of pixels to crop from the right.
+    pub right: usize,
+}
+
+impl FromStr for Crop {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s.split(':').collect::<Vec<_>>()[..] {
+            [left, right, top, bottom] => {
+                match (left.parse(), right.parse(), top.parse(), bottom.parse()) {
+                    (Ok(left), Ok(right), Ok(top), Ok(bottom)) => Ok(Crop {
+                        left,
+                        right,
+                        top,
+                        bottom,
+                    }),
+                    _ => Err(format!("Failed to parse crop region string: {}", s)),
+                }
+            }
+            _ => Err(format!("Bad resolution format: {}", s)),
+        }
+    }
+}
+
+impl serde::ser::Serialize for Crop {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::ser::Serializer,
+    {
+        let s = format!("{}:{}:{}:{}", self.left, self.right, self.top, self.bottom);
+
+        serializer.serialize_str(&s)
+    }
+}
+
+impl<'de> serde::de::Deserialize<'de> for Crop {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::de::Deserializer<'de>,
+    {
+        let s = <&str>::deserialize(deserializer)?;
+
+        Crop::from_str(s).map_err(serde::de::Error::custom)
+    }
+}
+
+#[derive(Default, Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct TransformProperties {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub fps: Option<u32>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub resolution: Option<crate::Resolution>,
```

Comment:
Why no docs for these?

Reply:
I didn't add these props but I can add the docs.